experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/MarquezContext.java

```diff
@@ -1,146 +1,92 @@
+        System.out.println(response2.getCode() + " " + response2.getReasonPhrase());
+        HttpEntity entity2 = response2.getEntity();
+        EntityUtils.consume(entity2);
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      log.error("Could not emit lineage", e);
     }
-    String namespace = protocol;
-    String name = input;
-    DatasetMeta meta = DbTableMeta.builder().sourceName(protocol).physicalName(name).build();
-    marquezClient.createDataset(namespace, name, meta);
-    return new DatasetId(namespace, name);
   }
 
-  public UUID startRun(long jobStartTime, List<String> inputs, List<String> outputs, String extraInfo) {
-    Set<DatasetId> inputIds = new HashSet<DatasetId>();
-    for (String input : inputs) {
-      inputIds.add(ensureDatasetExists(input));
-
-    }
-    Set<DatasetId> outputIds = new HashSet<DatasetId>();
-    for (String output : outputs) {
-      outputIds.add(ensureDatasetExists(output));
-    }
-    Map<String, String> context = ImmutableMap.of("spark.extraInfo", extraInfo);
-    marquezClient.createJob(jobNamespace, jobName, JobMeta.builder().inputs(inputIds).outputs(outputIds).type(BATCH).context(context ).build());
-    UUID runId = UUID.randomUUID();
-    marquezClient.createRun(jobNamespace, jobName, new RunMeta(runId.toString(), null, null, null));
-    marquezClient.markRunAsRunning(runId.toString(), Instant.ofEpochMilli(jobStartTime));
-    return runId;
-  }
-
-  public void success(UUID runId, int jobId, long jobEndTime, List<String> outputs) {
-    for (String output : outputs) {
-      updateOutput(output, runId);
-    }
-    Instant at = Instant.ofEpochMilli(jobEndTime);
-    marquezClient.markRunAsCompleted(runId.toString(), at);
-  }
-
-  public void failure(UUID runId, int jobId, long jobEndTime, List<String> outputs, Exception error) {
-    for (String output : outputs) {
-      updateOutput(output, runId);
-    }
-    Instant at = Instant.ofEpochMilli(jobEndTime);
-    error.printStackTrace(System.out);
-    marquezClient.markRunAsFailed(runId.toString(), at);
-  }
-
-  private void updateOutput(String output, UUID runId) {
-    String protocol;
-    try {
-      URL outputUrl = new URL(output);
-      protocol=outputUrl.getProtocol();
-    } catch (MalformedURLException e) {
-      protocol = "unknown";
-    }
-    String namespace = protocol;
-    String name = output;
-    DatasetMeta meta = DbTableMeta.builder().sourceName(protocol).physicalName(name).runId(runId.toString()).build();
-    marquezClient.createDataset(namespace, name, meta);
+  public static ObjectMapper createMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new JavaTimeModule());
+    mapper.setSerializationInclusion(Include.NON_NULL);
+    mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+    mapper.disable(DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE);
+    return mapper;
```

Comment:
I'm afraid that the objectmapper may end up conflicting with peoples dependencies

Reply:
What I've done in the past is shading Jackson. It's such a common dependency and it tends to be incompatible between versions. We certainly don't want to impact the user job in ant way. I had added shading in this project.